src/SingleArticleCard.jsx

```diff
@@ -65,8 +68,29 @@ function SingleArticleCard() {
             setCommentSuccess(true)
         })
         .catch((error) => {
-            console.error("Error posting comment", error)
-            alert("Unable to post comment")
+            console.error('Error posting comment', error)
+            alert('Unable to post comment')
+        })
+    };
+
+    function handleDeleteComment(comment_id, author) {  
+        if(user.username !== author){
+            setDeleteError('You can only delete your own comments!')
+            setDeleteSuccess(null);
+            return
+        }
+        deleteComment(comment_id)
+        .then(() => {
+            setComments((prevComments) =>
+            prevComments.filter((comment) => comment.comment_id !== comment_id)
+            );
+            setDeleteSuccess('Your comment was deleted');
+            setDeleteError(null);
+        })
+        .catch((error) => {
+            console.error('Error deleting comment', error);
+            setDeleteError('Could not delete comment. Please try again');
+            setDeleteSuccess(null);
```

Comment:
Good work here! There is no need to try and show this optimistically and the approach you've taken caters for dealing with a potential error, which works really well. Attempt the delete first, then update the state/show an error. Really nice!